src/api.js

```diff
@@ -2046,3 +2048,50 @@ api.declare({
     pendingTasks:   count,
   });
 });
+
+/** List worker-types for a given provisioner */
+api.declare({
+  method:     'get',
+  route:      '/provisioners/:provisionerId/worker-types',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name:       'listWorkerTypes',
+  stability:  API.stability.experimental,
+  output:     'list-workertypes-response.json#',
+  title:      'Get a list of all active worker-types',
+  description: [
+    'Get all active worker-types.',
+    '',
+    'The response is paged. If this end-point returns a `continuationToken`, you',
+    'should call the end-point again with the `continuationToken` as a query-string',
+    'option. By default this end-point will list up to 1000 worker-types in a single',
+    'page. You may limit this with the query-string parameter `limit`.',
+  ].join('\n'),
+}, async function(req, res) {
+  let continuation = req.query.continuationToken || null;
+  let limit = parseInt(req.query.limit || 1000, 10);
+
+  const provisionerId = req.params.provisionerId;
+  let workerTypes = await this.WorkerType.scan({provisionerId}, {continuation, limit});
+
+  let result = {
+    workerTypes: workerTypes.entries.map(workerType => workerType.workerType),
+  };
+
+  // If no taskGroup was found
+  if (!result.workerTypes.length) {
+    return res.reportError('ResourceNotFound',
```

Comment:
I think returning an empty array here would be OK.  At least, that would be accurate for provisioners that currently have no active workerTypes, even if it's inaccurate when `provisionerId` doesn't exist.  I suppose the most correct choice would be to consult the Provisioner table in this case, and return 404 if there's no such provisioner, otherwise [].  But I'm comfortable with just returning [] unconditionally, too.